File: InnerPod/Source/NSObject+IGListDiffable.m
//
//  NSObject+IGListDiffable.m
//  TestingProject
//
//  Created by Omar Zúñiga Lagunas on 28/04/20.
//

#import "NSObject+IGListDiffable.h"

@implementation NSObject (IGListDiffable)

- (id<NSObject>)diffIdentifier {
    return self;
}

- (BOOL)isEqualToDiffableObject:(id<IGListDiffable>)object {
    return [self isEqual:object];
}

@end

File: InnerPod/Source/NSObject+IGListDiffable.swift
//
//  NSObject+IGListDiffable.swift
//  TestingProject
//
//  Created by Omar Zúñiga Lagunas on 27/04/20.
//

import Foundation
import IGListKit

extension NSObject: ListDiffable {

  open func diffIdentifier() -> NSObjectProtocol {
    return self
  }

  open func isEqual(toDiffableObject object: ListDiffable?) -> Bool {
    return isEqual(object)
  }
}
